### PR TITLE
fix(components/tabs): push permalink query params via router

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2994359](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2994359)

### CODE REVIEW

File: libs/components/tabs/src/lib/modules/tabs/fixtures/tabset-loop.component.fixture.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
   [active]="activeIndex"
   (tabIndexesChange)="onTabIndexesChange($event)"
 >
-  @for (tab of tabArray; track tab) {
+  @for (tab of tabArray; track tab.tabHeading) {
```
Comment (johnhwhite) [inline libs/components/tabs/src/lib/modules/tabs/fixtures/tabset-loop.component.fixture.html:5]: This addresses a console warning.

File: libs/components/tabs/src/lib/modules/tabs/tabset-permalink.service.ts [hunk 2]
```diff
@@ -85,9 +70,15 @@ export class SkyTabsetPermalinkService implements OnDestroy {
       return;
     }
 
-    // Use `replaceState()` when the tabset is being initialized so an extra
-    // history item isn't added to the browser's back stack.
-    this.#location[initial ? 'replaceState' : 'go'](url);
+    // Navigate via the Angular router to update the URL and broadcast the
+    // change. Use `replaceUrl` on the initial navigation so an extra history
+    // item isn't added to the browser's back stack.
+    void this.#router.navigate([], {
```
Comment (johnhwhite) [inline libs/components/tabs/src/lib/modules/tabs/tabset-permalink.service.ts:76]: This is the real fix. Other changes are just code tidying.

File: libs/components/tabs/src/lib/modules/tabs/tabset.component.html [hunk 3]
```diff
@@ -19,7 +19,7 @@
           {{ dropdownTriggerButtonText }}
         </sky-dropdown-button>
         <sky-dropdown-menu>
-          @for (tabButton of tabButtons; track tabButton) {
+          @for (tabButton of tabButtons; track tabButton.buttonId) {
```
Comment (johnhwhite) [inline libs/components/tabs/src/lib/modules/tabs/tabset.component.html:22]: This addresses a console warning.

File: libs/components/tabs/src/lib/modules/tabs/tabset.component.html [hunk 4]
```diff
@@ -43,7 +43,7 @@
     }
   </span>
   <span class="sky-tabset-tabs">
-    @for (tabButton of tabButtons; track tabButton) {
+    @for (tabButton of tabButtons; track tabButton.buttonId) {
```
Comment (johnhwhite) [inline libs/components/tabs/src/lib/modules/tabs/tabset.component.html:46]: This addresses a console warning.

File: libs/components/tabs/src/lib/modules/tabs/tabset.component.spec.ts [hunk 5]
```diff
@@ -1814,6 +1814,8 @@ describe('Tabset component', () => {
 
       expect(replaceStateSpy).toHaveBeenCalledOnceWith(
         '/?foobar-active-tab=design-guidelines',
+        '',
+        jasmine.any(Object),
```
Comment (johnhwhite) [inline libs/components/tabs/src/lib/modules/tabs/tabset.component.spec.ts:1818]: Angular's router calls the same location replace method, but it passes two additional parameters.

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2828/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/forms-storybook/)
- [help-inline-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/help-inline-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/2828/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2828/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2828/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2828/playground/)

**Review by Blackbaud-SteveBrush — DISMISSED** [review]

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**